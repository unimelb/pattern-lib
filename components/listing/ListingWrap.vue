<template>
  <div :class="classes">
    <slot />
  </div>
</template>

<script>
export default {
  props: {
    cols: {
      type: String,
      default: '',
    },
    center: {
      type: Boolean,
      default: false,
    },
  },
  computed: {
    classes() {
      return {
        grid: true,
        'grid--center': this.center,
        [`grid--${this.cols}col`]: ['1', '2', '3', '4'].includes(this.cols),
      };
    },
  },
};
</script>
