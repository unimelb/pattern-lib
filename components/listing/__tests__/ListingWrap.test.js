import {
  shallow,
} from 'vue-test-utils';
import { axe, toHaveNoViolations } from 'jest-axe';
import ListingWrap from '../ListingWrap.vue';

expect.extend(toHaveNoViolations);

describe('ListingWrap', () => {
  it('should match snapshot', () => {
    const result = shallow(ListingWrap).element;
    expect(result).toMatchSnapshot();
  });

  it('Component throws no accessibility violations', (done) => {
    const html = shallow(ListingWrap).html();
    // pass anything that outputs html to axe
    return axe(html).then((response) => {
      expect(response).toHaveNoViolations();
      done();
    });
  });
});
