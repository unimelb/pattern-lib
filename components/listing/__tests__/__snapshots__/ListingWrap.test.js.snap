// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListingWrap should match snapshot 1`] = `
<div
  class="grid"
/>
`;
